internal/apps/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/dd-trace-go/internal/apps
 
-go 1.21
```

Comment:
just curious: was there a problem without this change?

Reply:
The `go mod tidy` I added in the Dockerfile was failing on smoke tests jobs running on go 1.19 and 1.20